File: HFSGuardaDiretorioVS2022/HFSGuardaDiretorio_CSharp/objetosdao/ExtensaoDAO.cs
/*
 * Created by SharpDevelop.
 * User: henrique.souza
 * Date: 06/07/2015
 * Time: 11:14
 * 
 * To change this template use Tools | Options | Coding | Edit Standard Headers.
 */
using System;
using System.Data;
using System.Data.SQLite;
using System.Collections.Generic;
using HFSGuardaDiretorio.comum;
using HFSGuardaDiretorio.objetos;

namespace HFSGuardaDiretorio.objetosdao
{
	/// <summary>
	/// Description of ExtensaoDAO.
	/// </summary>
	public sealed class ExtensaoDAO
	{
		private static ExtensaoDAO instancia = new ExtensaoDAO();
		
	    private SQLiteConnection conexao;

	    private SQLiteCommand cmd;
	    
	    private SQLiteDataReader dr = null;
		
		public static ExtensaoDAO Instancia {
			get {
				return instancia;
			}
		}
		
		private ExtensaoDAO()
		{
		}
		
	    private void fecharSqlPreparado() {
	        if (dr != null) {
	            dr.Close();
	        }
	    }
	
	    private Extensao novoObjeto(SQLiteDataReader dr) {
	        Extensao ret = new Extensao();
	        ret.Codigo = dr.GetInt32(0);
	        ret.Nome = dr.GetString(1);	        
	        ret.Bmp16 = (byte[]) dr.GetValue(2);
	        ret.Bmp32 = (byte[]) dr.GetValue(3);
	        
	        return ret;
	    }
	
	    public int consultarTotal() {
	        int total = 0;
	        conexao = Rotinas.getConexao();
			cmd = new SQLiteCommand("select count(*) from Extensoes", conexao);	
			dr = cmd.ExecuteReader();
	
	        if (dr.Read()) {
	            total = dr.GetInt32(0);
	        }
	        this.fecharSqlPreparado();
	
	        return total;
	    }
		
	    public List<Extensao> consultarTudo(IProgressoLog progressoLog) {
	        Extensao obj;
	        List<Extensao> lista = new List<Extensao>();
	        Progresso pb;
	        int i = 0;
	        int total = this.consultarTotal();
	
	        conexao = Rotinas.getConexao();
	        cmd = new SQLiteCommand("select cod, nome, bmp16, bmp32 from Extensoes", conexao);	
			dr = cmd.ExecuteReader();
	        	
	        if (total > 0) {
	            pb = new Progresso();
	            pb.Minimo = 0;
	            pb.Maximo = total - 1;
	            pb.Posicao = 0;
	            pb.Passo = 1;
	
	            while (dr.Read()) {
	                obj = novoObjeto(dr);
	                obj.Ordem = i + 1;
	
	                lista.Add(obj);
	
	                pb.Posicao = i;
	
	                if (progressoLog != null) {
	                    progressoLog.ProgressoLog(pb);
	                }
	
	                i++;
	            }
	        }
	
	        this.fecharSqlPreparado();
	
	        return lista;
	    }
		
	    public int incluir(Extensao obj) {
	        int ret;
	        conexao = Rotinas.getConexao();
	        cmd = new SQLiteCommand("insert into Extensoes(cod, nome, bmp16, bmp32) values(@1,@2,@3,@4)", conexao);	
			cmd.Parameters.Add("@1", DbType.Int32);
			cmd.Parameters.Add("@2", DbType.String, 20);
         	cmd.Parameters.Add("@3", DbType.Binary, 1000000);
         	cmd.Parameters.Add("@4", DbType.Binary, 1000000);
	        /*
	        cmd.Parameters.AddWithValue("@1", obj.Codigo);
	        cmd.Parameters.AddWithValue("@2", obj.Nome);
	        cmd.Parameters.AddWithValue("@3", obj.Bmp16);
	        cmd.Parameters.AddWithValue("@4", obj.Bmp32);
	        */	        
	       	cmd.Prepare();
	       	
	        cmd.Parameters["@1"].Value = obj.Codigo;
	        cmd.Parameters["@2"].Value = obj.Nome;
	        cmd.Parameters["@3"].Value = obj.Bmp16;
	        cmd.Parameters["@4"].Value = obj.Bmp32;
	        	        
	        ret = cmd.ExecuteNonQuery();
	        return ret;
	    }
	
	    public int excluir(int codigo) {
	        int ret;
	        conexao = Rotinas.getConexao();
				
	        cmd = new SQLiteCommand("delete from Extensoes where cod=@1", conexao);
	        cmd.Parameters.AddWithValue("@1", codigo);
	        cmd.Prepare();
	        ret = cmd.ExecuteNonQuery();
	        
	        return ret;
	    }
	
	    public int excluirTudo() {
	        int ret;
	        conexao = Rotinas.getConexao();
				
	        cmd = new SQLiteCommand("delete from Extensoes", conexao);
	        ret = cmd.ExecuteNonQuery();
	        
	        return ret;
	    }

		public int criarTabela() {
	        int ret;
	        conexao = Rotinas.getConexao();
	        cmd = new SQLiteCommand(
                "create table IF NOT EXISTS Extensoes("
                + "cod integer not null,"
                + "nome varchar(20) not null,"
                + "bmp16 BLOB COLLATE NOCASE null,"
                + "bmp32 BLOB COLLATE NOCASE null,"
                + "primary key (cod))", conexao);
	        ret = cmd.ExecuteNonQuery();
	        return ret;
	    }
		
	}
}

File: HFSGuardaDiretorioVS2022/HFSGuardaDiretorio_VBNet/objetosdao/ExtensaoDAO.vb
'
' * Created by SharpDevelop.
' * User: henrique.souza
' * Date: 06/07/2015
' * Time: 11:14
' * 
' * To change this template use Tools | Options | Coding | Edit Standard Headers.
' 

Imports System.Data
Imports System.Data.SQLite
Imports System.Collections.Generic
Imports HFSGuardaDiretorio.comum
Imports HFSGuardaDiretorio.objetos

Namespace objetosdao
	''' <summary>
	''' Description of ExtensaoDAO.
	''' </summary>
	Public NotInheritable Class ExtensaoDAO
		Private Shared m_instancia As New ExtensaoDAO()

		Private conexao As SQLiteConnection

		Private cmd As SQLiteCommand

		Private dr As SQLiteDataReader = Nothing

		Public Shared ReadOnly Property Instancia() As ExtensaoDAO
			Get
				Return m_instancia
			End Get
		End Property

		Private Sub New()
		End Sub

		Private Sub fecharSqlPreparado()
			If dr IsNot Nothing Then
				dr.Close()
			End If
		End Sub

		Private Function novoObjeto(dr As SQLiteDataReader) As Extensao
			Dim ret As New Extensao()
			ret.Codigo = dr.GetInt32(0)
			ret.Nome = dr.GetString(1)
			ret.Bmp16 = DirectCast(dr.GetValue(2), Byte())
			ret.Bmp32 = DirectCast(dr.GetValue(3), Byte())

			Return ret
		End Function

		Public Function consultarTotal() As Integer
			Dim total As Integer = 0
			conexao = Rotinas.getConexao()
			cmd = New SQLiteCommand("select count(*) from Extensoes", conexao)
			dr = cmd.ExecuteReader()

			If dr.Read() Then
				total = dr.GetInt32(0)
			End If
			Me.fecharSqlPreparado()

			Return total
		End Function

		Public Function consultarTudo(progressoLog As IProgressoLog) As List(Of Extensao)
			Dim obj As Extensao
			Dim lista As New List(Of Extensao)()
			Dim pb As Progresso
			Dim i As Integer = 0
			Dim total As Integer = Me.consultarTotal()

			conexao = Rotinas.getConexao()
			cmd = New SQLiteCommand("select cod, nome, bmp16, bmp32 from Extensoes", conexao)
			dr = cmd.ExecuteReader()

			If total > 0 Then
				pb = New Progresso()
				pb.Minimo = 0
				pb.Maximo = total - 1
				pb.Posicao = 0
				pb.Passo = 1

				While dr.Read()
					obj = novoObjeto(dr)
					obj.Ordem = i + 1

					lista.Add(obj)

					pb.Posicao = i

					If progressoLog IsNot Nothing Then
						progressoLog.ProgressoLog(pb)
					End If

					i += 1
				End While
			End If

			Me.fecharSqlPreparado()

			Return lista
		End Function

		Public Function incluir(obj As Extensao) As Integer
			Dim ret As Integer
			conexao = Rotinas.getConexao()
			cmd = New SQLiteCommand("insert into Extensoes(cod, nome, bmp16, bmp32) values(@1,@2,@3,@4)", conexao)
			cmd.Parameters.Add("@1", DbType.Int32)
			cmd.Parameters.Add("@2", DbType.[String], 20)
			cmd.Parameters.Add("@3", DbType.Binary, 1000000)
			cmd.Parameters.Add("@4", DbType.Binary, 1000000)
			'
'	        cmd.Parameters.AddWithValue("@1", obj.Codigo);
'	        cmd.Parameters.AddWithValue("@2", obj.Nome);
'	        cmd.Parameters.AddWithValue("@3", obj.Bmp16);
'	        cmd.Parameters.AddWithValue("@4", obj.Bmp32);
'	        

			cmd.Prepare()

			cmd.Parameters("@1").Value = obj.Codigo
			cmd.Parameters("@2").Value = obj.Nome
			cmd.Parameters("@3").Value = obj.Bmp16
			cmd.Parameters("@4").Value = obj.Bmp32

			ret = cmd.ExecuteNonQuery()
			Return ret
		End Function

		Public Function excluir(codigo As Integer) As Integer
			Dim ret As Integer
			conexao = Rotinas.getConexao()

			cmd = New SQLiteCommand("delete from Extensoes where cod=@1", conexao)
			cmd.Parameters.AddWithValue("@1", codigo)
			cmd.Prepare()
			ret = cmd.ExecuteNonQuery()

			Return ret
		End Function

		Public Function excluirTudo() As Integer
			Dim ret As Integer
			conexao = Rotinas.getConexao()

			cmd = New SQLiteCommand("delete from Extensoes", conexao)
			ret = cmd.ExecuteNonQuery()

			Return ret
		End Function

		Public Function criarTabela() As Integer
			Dim ret As Integer
			conexao = Rotinas.getConexao()
			cmd = New SQLiteCommand("create table IF NOT EXISTS Extensoes(" & "cod integer not null," & "nome varchar(20) not null," & "bmp16 BLOB COLLATE NOCASE null," & "bmp32 BLOB COLLATE NOCASE null," & "primary key (cod))", conexao)
			ret = cmd.ExecuteNonQuery()
			Return ret
		End Function

	End Class
End Namespace
